### DISCUSSION TITLE
Any chance to add surveys / polls to a discussion?

### DISCUSSION THREAD

**Discussion by timunie** [community/community#2967] [Discussions] [answered]
Image you want to have a feedback to some predefined options, e.g. which idea feature is most wanted. You may want to let the users choose in a poll-like view.

Here is an example:

> Hey folks, which color should our App have?
> - [ ] `Blue.....` ▓▓▓░░░░░░ 
> - [ ] `Red......` ▓░░░░░░░░ 
> - [x] `Green....` ▓▓▓▓▓▓▓░░

**Comment by konard** [comment]
We can also add emoji colors and numbers to the list of possible reactions.

* 🔴 item
* 🟠 item
* 🟢 item

or

* 0️⃣ item
* 1️⃣ item
* 2️⃣ item

**Comment by efonsecab** [comment]
I would like polls and surveys on discussions too.
Using reactions is good, but unfortunately too limited.

**Comment by nschloe** [comment]
This is in the making. From https://github.blog/2021-10-27-everything-new-from-universe-2021/:

> We’re also shipping two net-new features in the coming months:
>
> - Polls: Create polls to gauge interest on anything from features to interests to 
>          roadmap planning.

**Reply by efonsecab** [reply]
Thanks. That's good to know

**Comment by bryevdv** [comment]
Is there any update on the roadmap feature linked above, or more information about how it will work? Ideally we would use this for public votes from OSS project core team members, so it would be helpful if the respondents can be limited to a specific group somehow. Is there any chance of that possibility?

**Comment by spaltrowitz** [comment, marked as answer]
this has shipped! https://github.blog/changelog/2022-04-12-discussions-polls/

**Reply by Andre601** [reply]
Curious to know (Haven't tested it yet) but can you reference polls the same way as you can issues and pull requests?
Maybe even have the poll's current results displayed when linked in a separate line?

If either of those isn't a thing yet would I take the time to create a suggestion for it.

**Reply by bryevdv** [reply]
Also curious to know: is there a way to close a poll, or does closing the discussion close the poll? Is there a way to limit participants (e.g. to restrict to maintainers making a decision)

**Reply by Andre601** [reply]
Made a discussion to cover some of the things that just came into my mind:
https://github.com/github/feedback/discussions/14812

**Reply by gmourier** [reply]
Hey @spaltrowitz 👋

Nice addition to the discussion. Is there a way to add a poll to a discussion from another category than "Polls"?

Also, it would be great to be able to do this within issues and pull requests, sometimes technical discussions demanding choices appear within a pull request.

Thanks!

**Reply by Andre601** [reply]
Personally, I could imagine the usage of some markdown format to allow polls being made.
Maybe something using a code block with a specific code identifier, similar to the `suggestion` one in Pull request reviews?

F.e.
````markdown
```poll
Normal text is treated as the main question.

- [ ] Answers
- [ ] Use the checkbox format
- [ ] to make them stand out.
```
````

This would then render an actual poll for people to answer to.
I can see some great potential with this kind of setup.

**Reply by evi-liu** [reply]
> can you reference polls the same way as you can issues and pull requests?

Since a poll is just a discussion, you can reference it the same way as you do any other discussion. 

> is there a way to close a poll, or does closing the discussion close the poll?

You can lock the discussion to pause or end voting.

> Is there a way to add a poll to a discussion from another category than "Polls"?

Not today. Is this something you would find valuable?

**Reply by Andre601** [reply]
> Not today. Is this something you would find valuable?

This would certainly be a useful feature as maintainers may want to ask the community about topics outside of discussions. Like when someone makes a PR and with time a question comes up that a poll could answer, such as what coding practice should be used.
Pulling the people away from the PR into the discussions just for the sake of answering a Poll would IMO feel a bit bat design-wise.

**Reply by gmourier** [reply]
> Not today. Is this something you would find valuable?

Yes, for example at Meilisearch, we have discussions related to product / technical choices that are already well underway. Adding polls as the discussion progresses permits keeping the context within the initial discussion and avoiding sending people to another discussion.

Here is an [example in situation](https://github.com/meilisearch/product/discussions/394), this one is pretty simple since it's just about naming. Currently, we manage these with emojis but having the poll system would be beneficial, especially if we can add deadlines.

**Comment by Alexisbarraza** [comment]
Unsuscribe

**Reply by timunie** [reply]
If this was a question: 

Here you can do: 

![image](https://user-images.githubusercontent.com/47110241/163347052-3ca5e387-7927-4233-8815-023a36f1b237.png)

**Comment by alanhamlett** [comment]
https://github.com/alanhamlett/github-issue-polls
